libs/vm-modules/src/math/tensor/tensor_estimator.cpp

```diff
@@ -0,0 +1,159 @@
+
+#include <cstdint>
+#include <vector>
+
+using namespace fetch::vm;
+
+namespace fetch {
+namespace vm_modules {
+namespace math {
+
+using ArrayType  = fetch::math::Tensor<VMTensor::DataType>;
+using SizeType   = ArrayType::SizeType;
+using SizeVector = ArrayType::SizeVector;
+
+TensorEstimator::TensorEstimator(VMTensor &tensor)
+  : tensor_{tensor}
+{}
+
+ChargeAmount TensorEstimator::size()
+{
+  return low_charge;
+}
+
+ChargeAmount TensorEstimator::AtOne(TensorType::SizeType /*idx1*/)
+{
+  return low_charge;
+}
+
+ChargeAmount TensorEstimator::AtTwo(uint64_t /*idx1*/, uint64_t /*idx2*/)
+{
+  return low_charge;
+}
+
+ChargeAmount TensorEstimator::AtThree(uint64_t /*idx1*/, uint64_t /*idx2*/, uint64_t /*idx3*/)
+{
+  return low_charge;
+}
+
+ChargeAmount TensorEstimator::AtFour(uint64_t /*idx1*/, uint64_t /*idx2*/, uint64_t /*idx3*/,
+                                     uint64_t /*idx4*/)
+{
+  return low_charge;
+}
+
+ChargeAmount TensorEstimator::SetAtOne(uint64_t /*idx1*/, DataType const & /*value*/)
+{
+  return low_charge;
+}
+
+ChargeAmount TensorEstimator::SetAtTwo(uint64_t /*idx1*/, uint64_t /*idx2*/,
+                                       DataType const & /*value*/)
+{
+  return low_charge;
+}
+
+ChargeAmount TensorEstimator::SetAtThree(uint64_t /*idx1*/, uint64_t /*idx2*/, uint64_t /*idx3*/,
+                                         DataType const & /*value*/)
+{
+  return low_charge;
+}
+
+ChargeAmount TensorEstimator::SetAtFour(uint64_t /*idx1*/, uint64_t /*idx2*/, uint64_t /*idx3*/,
+                                        uint64_t /*idx4*/, DataType const & /*value*/)
+{
+  return low_charge;
+}
+
+ChargeAmount TensorEstimator::Fill(DataType const & /*value*/)
+{
+  return charge_func_of_tensor_size();
+}
+
+ChargeAmount TensorEstimator::FillRandom()
+{
+  return charge_func_of_tensor_size();
+}
+
+ChargeAmount TensorEstimator::Min()
+{
+  return charge_func_of_tensor_size();
+}
+
+ChargeAmount TensorEstimator::Max()
+{
+  return charge_func_of_tensor_size();
+}
+
+ChargeAmount TensorEstimator::Squeeze()
+{
+  return charge_func_of_tensor_size();
+}
+
+ChargeAmount TensorEstimator::Unsqueeze()
+{
+  return charge_func_of_tensor_size();
+}
+
+ChargeAmount TensorEstimator::Reshape(
+    fetch::vm::Ptr<fetch::vm::Array<TensorType::SizeType>> const &new_shape)
+{
+  FETCH_UNUSED(new_shape);
+  return charge_func_of_tensor_size();
+}
+
+ChargeAmount TensorEstimator::Sum()
+{
+  return charge_func_of_tensor_size();
+}
+
+ChargeAmount TensorEstimator::Transpose()
+{
+  return charge_func_of_tensor_size();
+}
+
+ChargeAmount TensorEstimator::FromString(fetch::vm::Ptr<fetch::vm::String> const &string)
+{
+  std::size_t val_size = 2;
+  return static_cast<ChargeAmount>(static_cast<std::size_t>(string->Length()) / val_size);
+}
+
+ChargeAmount TensorEstimator::ToString()
+{
+  return charge_func_of_tensor_size();
+}
+
+ChargeAmount TensorEstimator::charge_func_of_tensor_size(std::size_t factor)
```

Comment:
This needs a better name IMHO and the style is inconsistent. Maybe `ComputeChargeFromTensorSize()`?